src/listeners.js

```diff
@@ -0,0 +1,180 @@
+      }
+    }
+    this.listeners.push(listener);
+  }
+  
+  async reconcileConsumers() {
+    let Queues = [];
+    let allHooksData = [];
+    await this.Queues.scan(
+      {},
+      {
+        limit: 1000,
+        handler:(queue) => Queues.push(queue),
+      }
+    );
+    let continuationToken = undefined;
+    
+    do {
+      allHooksData  = await this.Hook.scan({}, {
+        limit: 1000,
+        continuation: continuationToken,
+      });
+      continuationToken = allHooksData.continuation;
+      allHooksData.entries.forEach(async (hook) => {
+        if (hook.bindings != []) {
+          const {hookGroupId, hookId} = hook;
+          const queue = _.find(Queues, {hookGroupId, hookId});
+          if (queue) {
+            const index = this.listeners.findIndex(({_queueName}) => listener._queueName === queue.queueName);
+            if (index == -1) {
+              debug('Existing queue..creating listener');
+              await this.createListener({hook, oldBindings: queue.bindings});
+            }
+            _.pull(Queues, queue);
+          } else {
+            debug('New queue..creating listener');
+            await this.createListener({hook, oldBindings: [{}]});
+            // Add to Queues table
+            debug('adding to Queues table');
+            await this.Queues.create({
+              hookGroupId,
+              hookId,
+              queueName: `${hookGroupId}/${hookId}`,
+              bindings: hook.bindings,
+            });
+          }
+        }
+      });
+    } while (continuationToken);
+    
+    // Delete the queues now left in the Queues list.
+    Queues.forEach(async (queue) => {     
+      if (!this.client.isFakeClient) {
+        if (await this.client.withChannel(async channel => channel.checkQueue(queue.queueName))) {
+          await this.client.withChannel(async channel => channel.deleteQueue(queue.queueName));
+        }
+      }
+      // Delete from this.listeners
+      let removeIndex = this.listeners.findIndex(({_queueName}) => queue.queueName === _queueName);
+      if (removeIndex > -1) {
+        const listener = this.listeners[removeIndex];
+        listener.stop();
+        this.listeners.splice(removeIndex, 1);
+      }
+      queue.remove();
+    });
+  }
+
+  async terminate() {
+    debug('Deleting all queues..');
+    let Queues = [];
+    await this.Queues.scan(
+      {},
+      {
+        limit: 1000,
+        handler:(queue) => Queues.push(queue),
+      }
+    );
+    Queues.forEach(async (queue) => {   
+      if (!this.client.isFakeClient) {  
+        if (await this.client.withChannel(async channel => channel.checkQueue(queue.queueName))) {
+          await this.client.withChannel(async channel => channel.deleteQueue(queue.queueName));
+        }
+      }
+      queue.remove();
+    });
+
+    // Will cancel all consumers
+    if (!this.client.isFakeClient) {
+      this.client.stop(); 
+    }
```

Comment:
The PulseListener class was given a running client, so it's a little awkward for that instance to then close the client -- what if something else was also using that client?  I admit, nothing else *is* using it right now, but that might not be the case forever.

So I think that you can omit calling `this.client.stop()`.